main.rb

```diff
@@ -40,116 +40,134 @@ def my_select
+            else check(true, param)
+            end
+    state == true ? state : false
   end
 
   def my_all?(param = true)
     if block_given?
       each { |item| return false unless yield(item) }
-    elsif param.is_a? Regexp
-      each { |item| return false unless param.match(item) }
-    elsif param.is_a? Class
-      each { |item| return false unless item.instance_of? param }
-    else
-      each { |item| return false unless item == param }
+      return true
     end
-    true
+    state = case param
+            when Regexp then check_regexp(false, param, true)
+            when Class then check_class(false, param, true)
+            else check(false, param, true)
+            end
+    state == false ? state : true
   end
 
   def my_none?(param = true)
     if block_given?
       each { |item| return false if yield(item) }
-    elsif param.is_a? Regexp
-      each { |item| return false if param.match(item) }
-    elsif param.is_a? Class
-      each { |item| return false if item.instance_of? param }
-    else
-      each { |item| return false if item == param }
+      return true
+    end
+    state = case param
+            when Regexp then check_regexp(false, param)
+            when Class then check_class(false, param)
+            else check(false, param)
+            end
+    state == false ? state : true
+  end
+
+  def check_regexp(status, param, negative = false)
+    each { |item| return status unless param.match(item) } if negative
+    each { |item| return status if param.match(item) } unless negative
+  end
+
+  def check_class(status, param, negative = false)
+    each { |item| return status unless item.instance_of? param } if negative
+    each { |item| return status if item.instance_of? param } unless negative
+  end
+
+  def check(status, param, negative = false)
+    each do |item|
+      cond = if param == true
+               !item.nil? && item
+             else
+               item == param
+             end
+
+      cond = negative ? !cond : cond
+
+      return status if cond
     end
-    true
   end
 
   def my_count(cond = nil)
     counter = 0
     each do |item|
-      counter += 1 if (block_given? && yield(item)) || (item == cond) || !cond
+      counter += 1 if (block_given? && yield(item)) || (item == cond) || (!cond && !block_given?)
     end
     counter
   end
 
   def my_inject(*args)
-    acc = 0
-    acc = args[0] if !args[0].is_a?(Symbol) && !args.empty?
-
-    return multiply_els(self) if args.include? :*
+    acc = setup_acc(args)
+    return multiply_els(self, acc) if args.include? :*
 
-    return add_els(self) if args.include? :+
+    return add_els(self, acc) if args.include? :+
 
-    return divide_els(self) if args.include? :/
+    return divide_els(self, acc) if args.include? :/
 
-    return subs_els(self) if args.include? :-
+    return subs_els(self, acc) if args.include? :-
 
+    each { |item| acc = yield(acc, item) }
+    acc
+  end
```

Comment:
#### :bug: As @maelfosso explained earlier. `my_inject` should be able to handle basic operations like '+', '/', '*', '-' and so on. See examples with the comparison of built-in methods below.

- `my_inject` should be able to accept a block with different operations:

   ```ruby
  # Multiplication
   irb(main):050:0> [1,2,3].inject{ |prod, n| prod * n }
   => 6
   irb(main):051:0> [1,2,3].my_inject{ |prod, n| prod * n }
   => ""
  # Addition
   irb(main):050:0> [1,2,3].inject{ |sum, n| sum + n }
   => 6
   irb(main):051:0> [1,2,3].my_inject{ |sum, n| sum + n }
   TypeError (no implicit conversion of Integer into String)
   ```

### Find more details about [`inject?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject)